zamba/data/video.py

```diff
@@ -325,6 +325,41 @@ def validate_total_frames(cls, values):
         return values
 
 
+def get_cached_array_path(config, vid_path, cache_dir):
+    """Get the path to where the cached array would be, if it exists.
+
+    config: dictionary or VideoLoaderConfig
+    vid_path: string path to the video, or Path, or S3Path
+    cache_dir: string directory name, or Path, or S3Path
+
+    returns: Path object to the cached data
+    """
+    if isinstance(config, VideoLoaderConfig):
+        # NOTE: config.dict() and dict(config) are equal, but their string representations are not
+        # so we should use config.dict() everywhere
+        config = config.dict()
+
+    # don't include `cleanup_cache` or `cache_dir` in the hashed config
+    # NOTE: sorting the keys avoids a cache miss if we see the same config in a different order
+    keys = config.keys() - {"cleanup_cache", "cache_dir"}
+    hashed_part = {k: config[k] for k in sorted(keys)}
```

Comment:
maybe a touch clearer but basically equivalent. fine to leave as is if you prefer
```
excluded_keys = ["cleanup_cache", "cache_dir"]
hashed_part = {k: config[k] for k in sorted(keys) if k not in excluded_keys}
```

Reply:
Set operations on dictionary keys are new; I think it's a good idiom to add to our repertoire.